core/common/src/main/java/alluxio/AlluxioURI.java

```diff
@@ -132,10 +144,23 @@ public boolean equals(Object o) {
    * @return the authority, null if it does not have one
    */
   @Nullable
-  public String getAuthority() {
+  public Authority getAuthority() {
     return mUri.getAuthority();
   }
 
+  /**
+   * @return the authority type of the {@link AlluxioURI}
+   */
+  public AuthorityType getAuthorityType() {
+    Authority authority = mUri.getAuthority();
+    if (authority == null) {
+      return AuthorityType.NONE;
+    } else if (authority instanceof ZookeeperAuthority) {
+      return AuthorityType.ZOOKEEPER;
+    }
+    return AuthorityType.HOSTNAME_PORT;
```

Comment:
what if it is not in the form of `host:port` (e.g., `host:a:b`), is it still valid?

Reply:
@apc999 would it be better to change the `AuthorityType` from` HOSTNAME_PORT` to `NORMAL` or `MASTER`

Reply:
is PORT required? e.g. is `host` a valid authority? if not required, I would simply call it HOST. Otherwise, HOSTNAME_PORT is good IMO.

Reply:
RE: "HOSTNAME_PORT authority refers to all the other authorities other than ZookeeperAuthority."

This doesn't sound right - each different type of authority should be modeled as its own type. If we support `host1:port1,host2:port2` in the future, I would expect us to introduce a new type of authority.

Reply:
Yes, It doesn't sound right, we could make it more specific later if we support multiple master
-> ZOOKEEPER, SINGLE_MASTER, MULTI_MASTER, OTHER, NO_AUTH

Reply:
The current AlluxioURITest including authority like "localhost" "localhost:19998". I am not sure if we use AlluxioURI for authority without port?

Reply:
@LuQQiu Then I would suggest to go with more specific here:

```
if (authority == null) {
   return AuthorityType.NONE;
} else if (authority instanceof ZookeeperAuthority) {
  return AuthorityType.ZOOKEEPER;
} else (authority instanceof HostnamePortAuthority) {
  return AuthorityType.HOSTNAME_PORT; // or SINGLE_MASTER if you like it better
} else {
  return AuthorityType.OTHER
}
```